### PR TITLE
P2166R1 A Proposal to Prohibit std::basic_string and std::basic_strin…

### LINKED ISSUE #4651: [2021-06 LWG Motion 14] P2166R1 A Proposal to Prohibit std::basic_string and std::basic_string_view construction from nullptr
[P2166R1](https://wg21.link/p2166r1)

### PR DESCRIPTION
**Pull request by jensmaurer**
…g_view construction from nullptr

Fixes #4651
Fixes cplusplus/papers#887

### CODE REVIEW

File: source/strings.tex [hunk 1]
```diff
@@ -824,6 +824,7 @@
       constexpr explicit basic_string(const T& t, const Allocator& a = Allocator());
     constexpr basic_string(const charT* s, size_type n, const Allocator& a = Allocator());
     constexpr basic_string(const charT* s, const Allocator& a = Allocator());
+    constexpr basic_string(nullptr_t) = delete;
```
Comment (jwakely) [inline source/strings.tex:827]: I should have asked this during the review, but why are these `constexpr`?

Comment (tkoeppe) [reply source/strings.tex:827]: "constexpr all the things" is a life style! :-)

Comment (CaseyCarter) [reply source/strings.tex:827]: I thought this was odd until the contributor who prepared our PR for this decided to strengthen `noexcept` as well and raised my bar for weird.

Comment (jensmaurer) [reply source/strings.tex:827]: I think neither `constexpr` nor `noexcept` make sense / changes anything for deleted functions.

Comment (jwakely) [reply source/strings.tex:827]: I've created #4706 for this.

Comment (JohelEGP) [reply source/strings.tex:827]: I thought it made for a nice alignment.

Comment (tkoeppe) [reply source/strings.tex:827]: @jwakely there you have it ^^^ :-)

**Pull request merged**